File: Algorithms/Implementation/Service Lane.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Result
{
    /*
     * Complete the 'serviceLane' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. 2D_INTEGER_ARRAY cases
     *  3. INTEGER_ARRAY width
     */
    public static List<int> serviceLane(int n, List<List<int>> cases, List<int> width)
    {
        List<int> results = new List<int>();

        foreach (var c in cases)
        {
            int start = c[0];
            int end = c[1];
            int minWidth = int.MaxValue;

            for (int i = start; i <= end; i++)
            {
                if (width[i] < minWidth)
                {
                    minWidth = width[i];
                }
            }

            results.Add(minWidth);
        }

        return results;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        string? outputPath = Environment.GetEnvironmentVariable("OUTPUT_PATH");
        if (outputPath == null)
        {
            Console.WriteLine("OUTPUT_PATH environment variable is not set.");
            return;
        }

        using (TextWriter textWriter = new StreamWriter(outputPath, true))
        {
            string[] firstMultipleInput = Console.ReadLine().TrimEnd().Split(' ');

            int n = Convert.ToInt32(firstMultipleInput[0]);
            int t = Convert.ToInt32(firstMultipleInput[1]);

            List<int> width = Console.ReadLine().TrimEnd().Split(' ').Select(int.Parse).ToList();

            List<List<int>> cases = new List<List<int>>();

            for (int i = 0; i < t; i++)
            {
                cases.Add(Console.ReadLine().TrimEnd().Split(' ').Select(int.Parse).ToList());
            }

            List<int> result = Result.serviceLane(n, cases, width);

            textWriter.WriteLine(String.Join("\n", result));
        }
    }
}

File: Algorithms/Implementation/Service Lane.php
<?php

/*
 * Complete the 'serviceLane' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. 2D_INTEGER_ARRAY cases
 */

function serviceLane($width, $cases) {
    # Write your code here
    global $width;
    $results = [];

    foreach ($cases as $case) {
        $start = $case[0];
        $end = $case[1];
        $minWidth = PHP_INT_MAX;

        for ($i = $start; $i <= $end; $i++) {
            if ($width[$i] < $minWidth) {
                $minWidth = $width[$i];
            }
        }

        $results[] = $minWidth;
    }
    return $results;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));
$n = intval($first_multiple_input[0]);
$t = intval($first_multiple_input[1]);

$width_temp = rtrim(fgets(STDIN));
$width = array_map('intval', preg_split('/ /', $width_temp, -1, PREG_SPLIT_NO_EMPTY));
$cases = array();

for ($i = 0; $i < $t; $i++) {
    $cases_temp = rtrim(fgets(STDIN));
    $cases[] = array_map('intval', preg_split('/ /', $cases_temp, -1, PREG_SPLIT_NO_EMPTY));
}

$result = serviceLane($n, $cases);

fwrite($fptr, implode("\n", $result) . "\n");
fclose($fptr);
